src/traits/data.jl

```diff
@@ -136,17 +136,111 @@ Tables.materializer(D::Type{<:Data}) = D
 # DATAFRAME INTERFACE
 # --------------------
 
-function Base.getproperty(data::Data, col::Symbol)
-  if col == :geometry
+function Base.getproperty(data::Data, var::Symbol)
+  if var == :geometry
     domain(data)
   else
     cols = Tables.columns(values(data))
-    Tables.getcolumn(cols, col)
+    Tables.getcolumn(cols, var)
   end
 end
 
-Base.getproperty(data::Data, col::AbstractString) =
-  getproperty(data, Symbol(col))
+Base.getproperty(data::Data, var::AbstractString) =
+  getproperty(data, Symbol(var))
+
+function Base.getindex(data::Data, inds::AbstractVector{Int}, vars::AbstractVector{Symbol})
+  _checkvars(vars)
+  _rmgeometry!(vars)
+  table = values(data)
+
+  newdom   = view(domain(data), inds)
+  subset   = Tables.subset(table, inds)
+  cols     = Tables.columns(subset)
+  pairs    = (var => Tables.getcolumn(cols, var) for var in vars)
+  newtable = (; pairs...) |> Tables.materializer(table)
+
+  vals = Dict(paramdim(newdom) => newtable)
+  constructor(data)(newdom, vals)
```

Comment:
Let's be more systematic here and define dom = domain(data), tab = values(data), then define newdom = view(dom, inds), newtab = ...